Training/Training/Training.c

```diff
@@ -1,14 +1,161 @@
+void BufferDrain () {
+   while (getchar () != '\n');
+}
+
+/// <summary>Function to check if the given input is a phrase or a number</summary>
+bool IsValidNum (char* phrase, int* num) {
+   errno = 0;
+   char* endptr = NULL;
+   *num = strtol (phrase, &endptr, 10);
+   return endptr == phrase || *endptr != '\0' || errno == ERANGE || phrase[0] == ' ' || phrase[0] == '\t' ?
+      false : true;
+}
+
+/// <summary>Function to run test cases</summary>
+void TestCases () {
+   int Length;
+   char* Inputs[] = { "Was it a car or a cat I saw?", "I did, did I?", "Don't nod", "TRUMPF",
+      "Metamation", "121", "232", "123454321", "123456789", "0", "!@#  $%", " ", "-878" }, * Expected[] = {
+      "Palindrome", "Palindrome", "Palindrome", "Not a Palindrome", "Not a Palindrome","Palindrome",
+      "Palindrome", "Palindrome", "Not a Palindrome", "Palindrome", "Invalid", "Invalid", "Not a Palindrome" },
+      * output = NULL;
+   Length = sizeof (Inputs) / sizeof (Inputs[0]);
+   printf (ANSI_COLOR_YELLOW"\n-------------------------------------------------------------------"
+           "--------------\n|------------Input-------------|-----Expected-----|------Output------"
+           "|--Result--|"ANSI_COLOR_RESET);
+   for (int i = 0; i < Length; i++) {
+      int num, pChecker;
+      long long int rev = 0;
+      pChecker = IsValidNum (Inputs[i], &num) ?
+         ((NumberReverser (num) == num && num >= 0) ? PALINDROME : NOTPALINDROME) :
+         PalindromeChecker (Inputs[i]);
+      output = pChecker == PALINDROME ? "Palindrome" : pChecker == NOTPALINDROME ?
+         "Not a Palindrome" : "Invalid";
+      printf ("\n  %-31s%-19s%-20s%s", Inputs[i], Expected[i], output,
+              strcmp (output, Expected[i]) == 0 ? ANSI_COLOR_GREEN"Passed"ANSI_COLOR_RESET :
+              ANSI_COLOR_RED"Failed"ANSI_COLOR_RESET);
+   }
+   printf (ANSI_COLOR_YELLOW"\n-------------------------------------------------------------------"
+           "--------------\n"ANSI_COLOR_RESET);
+}
+
+/// <summary>Function to display the appropriate message and/or the reversed number</summary>
+void Display (int output, int isNum, long long int rev) {
+   isNum == 0 ? printf (output == PALINDROME ? "Palindrome\n" : output == INVALID ?
+                        "Enter a valid string!\n" : "Not a palindrome!\n") :
+      output == PALINDROME ? printf ("\nReversed: %lld\nPalindrome\n", rev) :
+      printf ("\nReversed: %lld\nNot a palindrome!\n", rev);
+}
+
+/// <summary>Function to accept input from the user</summary>
+void UserInput () {
+   char phrase[MAX], * result = NULL;
+   fgets (phrase, MAX, stdin);
+   result = strchr (phrase, '\n');
+   if (result == NULL) {
+      printf ("Enter a valid string!\n");
+      BufferDrain ();
+   } else {
+      int output, isNum = 1, num;
+      long long int rev = 0;
+      phrase[strlen (phrase) - 1] = '\0';
+      IsValidNum (phrase, &num) ? (output = (rev = NumberReverser (num)) == num && num >= 0 ?

```

Comment:
Ternary operator is best used for assignment statements. In cases like this, having a separate if () helps us to understand the code a little better. 
```suggestion
     if (IsValidNum (phrase, &num))
 output = (rev = NumberReverser (num)) == num && num >= 0 ? PALINDROME : NOTPALINDROME;

```